File: java/java_study/src/section01/Solution02.java
package section01;

public class Solution02 {

    /*
    * 대소문자 변환
    *
    * 대문자와 소문자가 같이 존재하는 문자열을 입력받아 대문자는 소문자로 소문자는 대문자로 변환하여 출력하는 프로그램
    *
    * input : StuDY
    * output : sTUdy
    * */

    public static void main(String[] args) {
        System.out.println("result : " + solution("StuDY"));
    }

    public static String solution(String str) {
        String answer = "";

        // isUpperCase() | toUpperCase()
//        for (char c : str.toCharArray()) {
//            if (Character.isUpperCase(c)) {
//                answer += Character.toLowerCase(c);
//            } else {
//                answer += Character.toUpperCase(c);
//            }
//        }

        // ASCII
        // 대문자 : 65 ~ 90
        // 소문자 : 97 ~ 122
        for (char c : str.toCharArray()) {
            if (c >= 65 && c <= 90) {
                answer += (char) (c + 32);
            } else {
                answer += (char) (c - 32);
            }
        }

        return answer;
    }

}

File: javascript/section03/solution02.js
/*
유효한 팰린드롬

앞에서 읽을때나 뒤에서 읽을때나 같은 문자열을 팰린드롬이라고 한다.
문자열이 입력되면 해당 문자열이 팰린드로이라면 "YES", 아니라면 "NO"를 출력하는 프로그램 작성
단 회문 검사할 때 알파벳만 가지고 회문을 검사하며, 대소문자를 구분하지 않음

input : found7, time: study; Yduts; emit, 7Dnuof
output : YES
*/

let str = "found7, time: study; Yduts; emit, 7Dnuof";
console.log(`result : ${solution(str)}`);

function solution(str) {
    let answer = "YES";

    // /[^a-z]/g 소문자가 아닌것들을 '' 제거
    str = str.toLowerCase().replace(/[^a-z]/g, '');
    // split('') 문자배열화해서, reverse() 뒤집은 다음, join() 다시 문자열로 이어서 검증
    if(str.split('').reverse().join('') !== str) answer = "NO";

    return answer;
}
